diff -uNr lua-5.1.5/src/loadlib.c lua-5.1.5_patched/src/loadlib.c
--- lua-5.1.5/src/loadlib.c	2009-09-09 15:17:16.000000000 +0200
+++ lua-5.1.5_patched/src/loadlib.c	2020-06-02 21:46:02.752869142 +0200
@@ -60,6 +60,30 @@
 
 #include <dlfcn.h>
 
+
+#undef setprogdir
+
+
+/*
+** Replace in the path (on the top of the stack) any occurrence
+** of LUA_EXECDIR with the executable's path.
+*/
+static void setprogdir (lua_State *L) {
+  Dl_info tDlInfo;
+  int iResult;
+  char *fname, *lb;
+  iResult = dladdr(setprogdir, &tDlInfo);
+  if (iResult == 0 || tDlInfo.dli_fname == NULL || (lb = strrchr(tDlInfo.dli_fname, '/')) == NULL )
+    luaL_error(L, "unable to get ModuleFileName");
+  else {
+    fname = strndup(tDlInfo.dli_fname, lb-tDlInfo.dli_fname);
+    luaL_gsub(L, lua_tostring(L, -1), LUA_EXECDIR, fname);
+    free(fname);
+    lua_remove(L, -2);  /* remove original string */
+  }
+}
+
+
 static void ll_unloadlib (void *lib) {
   dlclose(lib);
 }
