diff -uNr lua-5.4.3/src/luaconf.h lua-5.4.3_patched/src/luaconf.h
--- lua-5.4.3/src/luaconf.h	2021-03-15 14:32:52.000000000 +0100
+++ lua-5.4.3_patched/src/luaconf.h	2021-06-21 09:56:34.085731294 +0200
@@ -324,6 +324,8 @@
 ** ===================================================================
 */
 
+#define LUA_COMPAT_5_3
+
 /*
 @@ LUA_COMPAT_5_3 controls other macros for compatibility with Lua 5.3.
 ** You can define it to get all options, or change specific options
